core/src/main/java/hudson/security/SecurityRealm.java

```diff
@@ -290,15 +290,44 @@ public void doLogout(StaplerRequest req, StaplerResponse rsp) throws IOException
         Authentication auth = SecurityContextHolder.getContext().getAuthentication();
         SecurityContextHolder.clearContext();
 
-        // reset remember-me cookie
-        Cookie cookie = new Cookie(ACEGI_SECURITY_HASHED_REMEMBER_ME_COOKIE_KEY,"");
+        String contextPath = req.getContextPath().length() > 0 ? req.getContextPath() : "/";
+        resetRememberMeCookie(req, rsp, contextPath);
+        clearStaleSessionCookies(req, rsp, contextPath);
+
+        rsp.sendRedirect2(getPostLogOutUrl(req,auth));
+    }
+
+    private void resetRememberMeCookie(StaplerRequest req, StaplerResponse rsp, String contextPath) {
+        Cookie cookie = new Cookie(ACEGI_SECURITY_HASHED_REMEMBER_ME_COOKIE_KEY, "");
         cookie.setMaxAge(0);
         cookie.setSecure(req.isSecure());
         cookie.setHttpOnly(true);
-        cookie.setPath(req.getContextPath().length()>0 ? req.getContextPath() : "/");
+        cookie.setPath(contextPath);
         rsp.addCookie(cookie);
+    }
 
-        rsp.sendRedirect2(getPostLogOutUrl(req,auth));
+    // See https://github.com/jenkinsci/extras-executable-war/blob/6558df699d1366b18d045d2ffda3e970df377873/src/main/java/Main.java#L79-L97
+    private static final String JSESSIONID_COOKIE_NAME =
+            System.getProperty("executableWar.jetty.sessionIdCookieName");
+    private static final boolean DISABLE_CUSTOM_JSESSIONID_COOKIE_NAME =
+            Boolean.getBoolean("executableWar.jetty.disableCustomSessionIdCookieName");
+
+    private void clearStaleSessionCookies(StaplerRequest req, StaplerResponse rsp, String contextPath) {
+        final String cookieName;
+        if (!DISABLE_CUSTOM_JSESSIONID_COOKIE_NAME && JSESSIONID_COOKIE_NAME != null) {
+            cookieName = JSESSIONID_COOKIE_NAME;
+        } else {
+            cookieName = "JSESSIONID.";
```

Comment:
By default Jetty would set just `JSESSIONID` as a cookie name. So maybe dot should be removed from here

Reply:
So, thinking about this, I'm not sure it's the right behavior.

My goal is to delete all the stale cookies.

1. Let's say you have a server which was configured with cookieName with the jenkins default (`JSESSIONID.`+_random_)
2. User visits and server and over time (as it's restarted or whatever) collects cookies.
3. Someone turns off the feature (and restarts jenkins).
4. Now you have a whole bunch of `JSESSIONID.`+_random_ cookies which will never go away.
5. User is stuck and tries to log out -- this feature will not help them if we only look at the current configuration.

Remember, the goal is not to delete the current jenkins session cookie, jenkins is already managing (and **using**) that cookie. The goal is to delete all the other ones.

Reply:
For my reference, the correct code to try to emulate the underlying behavior would have been:
```
        if (!DISABLE_CUSTOM_JSESSIONID_COOKIE_NAME) {
            if (JSESSIONID_COOKIE_NAME != null) {
                cookieName = JSESSIONID_COOKIE_NAME;
            } else {
                cookieName = "JSESSIONID.";
            }
        } else {
            cookieName = "JSESSIONID";
        }
```
Or to just use the `Field` setting code to look up the value, but, as noted above, that's pointless, as it doesn't accomplish the actual goal here.